### ISSUE TITLE
`TypeError` from passing invalid arguments when there's more than one query present

### ISSUE THREAD

**Issue by dburles** [slicknode/graphql-query-complexity#61]
A query such as:

```
query {
  requiredArgs(count: x) {
    scalar
    complexScalar
  }
  nonNullItem {
    scalar
    complexScalar
    variableScalar(count: 10)
  }
}
```

results in [complexityMap](https://github.com/slicknode/graphql-query-complexity/blob/master/src/QueryComplexity.ts#L461), being `undefined`, and the error `TypeError: Cannot convert undefined or null to object` arising [from this call](https://github.com/slicknode/graphql-query-complexity/blob/master/src/QueryComplexity.ts#L465).

Here's a reproduction https://github.com/dburles/graphql-query-complexity/blob/bug/src/__tests__/QueryComplexity-test.ts#L800-L821. Running this test will result in the error being thrown.

GraphQL should be reporting `"Argument \"count\" has invalid value x."`.

**Comment by dburles** [comment]
Hey @ivome, I've done some digging this morning, and I believe the underlying cause of this problem is that [validation occurs in parallel](https://github.com/graphql/graphql-js/blob/main/src/validation/validate.ts#L75). A query complexity rule cannot be used as if it were the last rule in the chain, as the rules are not ordered.

```js
// Results in `TypeError: Cannot convert undefined or null to object`
validate(schema, document, [
  ...specifiedRules,
  createComplexityRule({ ... })
]);
```

I've updated the reproduction to show this occurring: https://github.com/dburles/graphql-query-complexity/blob/bd05756ddf52c57ddfa2ca4e65a528b05fa3480c/src/__tests__/QueryComplexity-test.ts#L822-L832

I believe then that you may need to make changes to accomodate `complexityMap` being `undefined`.

**Comment by dburles** [comment]
Thanks @ivome, this looks great! I'll have a look and see if there may be any additional test cases worth adding.

**Comment by dburles** [comment]
I've been thinking about adapting some test cases from the core GraphQL validator tests, and in reality I don't think it's scalable. Instead, the rule should be defensive and ensure that it's only acting on values that are present and are the expected type. There's an infinite variety of incorrectly formatted queries.

**Comment by ivome** [comment]
This is in a way what happens now. With the implemented fix, invalid values should raise a meaningful error message at least. This should be identical to the one that would be raised by the GraphQL library, since it uses the same function internally and just passes the error. 

If you find any other invalid queries or edge cases that we need to address, feel free to open another ticket or PR. I'll close this here as the original issue is fixed.
